lib/puppet/property.rb

```diff
@@ -346,6 +362,21 @@ def insync?(is)
     end
   end
 
+  # This method tests if two values are insync? outside of the properties current
+  # should value. This works around the requirement for corrective_change analysis
+  # that requires two older values to be compared with the properties potentially
+  # custom insync? code.
+  #
+  # @param [Object] orig_should the original should value
+  # @param [Object] orig_is the original is value
+  # @return [Boolean] whether or not the values are in sync or not
+  # @api private
+  def insync_values?(orig_should, orig_is)
+    test_property = self.dup
```

Comment:
Lol. Its a pita. I can always right a dpittman-style essay on why its bad in the comments :-).

I'll try to do the other methodology, see how far I get, at least try to understand why it doesn't work. Having it contained like you state feels better than doing it from-a-distance at least.